### ISSUE TITLE
MagickWand Gravity + composite

### ISSUE THREAD

**Issue by Danack** [ImageMagick/ImageMagick#13]
Hi,

Is gravity supposed to affect image composition through the MagickWand API? e.g. for the example code below.

This is an attempt to have the same effect as `convert -composite -gravity center background.png overlay.png result.png`.

Currently it does not appear to be as the output of the code is:
![gravity](https://cloud.githubusercontent.com/assets/1505719/8325902/81c40a44-1a4c-11e5-93e6-644e498fc3c0.png)

With the composited image in the top left, as opposed to in the centre as expected.

```

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <malloc.h>
#include <wand/MagickWand.h>


int main(int argc,char **argv) {
    MagickWand *magick_wand_background;
    MagickWand *magick_wand_foreground;
    MagickWandGenesis();
    
    magick_wand_background = NewMagickWand();
    magick_wand_foreground = NewMagickWand();

    MagickReadImage(magick_wand_background, "magick:logo");
    MagickReadImage(magick_wand_foreground, "magick:rose");
    
 
    MagickSetGravity(magick_wand_background, CenterGravity);
    MagickSetGravity(magick_wand_foreground, CenterGravity);
 
    MagickCompositeImage(
        magick_wand_background,
        magick_wand_foreground,
        AtopCompositeOp,
        0, 0
    );
    
    MagickSetFormat(magick_wand_background, "png");

    MagickWriteImage(magick_wand_background, "./output/gravity.png");
    MagickWandTerminus();

    return 0;
}

```

**Comment by dlemstra** [comment]
You should calculate x and y instead of passing 0,0 as arguments for MagickCompositeImage. You can see how we do that in 'wand/mogrify.c'.

**Comment by Danack** [comment]
Hi Dlemstra,

The reason I'm asking is that it seems a lot of people who are taking command line scripts like `convert -composite -gravity center background.png overlay.png result.png` and want to call them programmatically through Imagick. Those people are expecting that the setGravity operation would affect the composite operation.

If setting the gravity through the Magick API isn't supposed to affect the composite operations, I could document that. In which case, which operations are affected by setting the gravity through the Magick API?

**Comment by urban-warrior** [comment]
We have a couple of helper MagickCore methods that are responsive to geometries such as ParseGravityGeometry() and GravityAdjustGeometry().  We need to export them to the MagickWand API.  With these methods, you can return the geometry based on gravity before you call the composite method.

**Comment by dlemstra** [comment]
We decided to add a new method to Wand that will be available in ImageMagick 6.9.1-7.

```C
MagickBooleanType MagickCompositeImageGravity(MagickWand *wand,
  const MagickWand *source_wand,const CompositeOperator compose,
  const GravityType gravity)
```

So you will be able to do this:
```C
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <malloc.h>
#include <wand/MagickWand.h>


int main(int argc,char **argv) {
    MagickWand *magick_wand_background;
    MagickWand *magick_wand_foreground;
    MagickWandGenesis();

    magick_wand_background = NewMagickWand();
    magick_wand_foreground = NewMagickWand();

    MagickReadImage(magick_wand_background, "magick:logo");
    MagickReadImage(magick_wand_foreground, "magick:rose");

    MagickCompositeImageGravity(
        magick_wand_background,
        magick_wand_foreground,
        AtopCompositeOp,
        CenterGravity
    );

    MagickSetFormat(magick_wand_background, "png");

    MagickWriteImage(magick_wand_background, "./output/gravity.png");
    MagickWandTerminus();

    return 0;
}
```
